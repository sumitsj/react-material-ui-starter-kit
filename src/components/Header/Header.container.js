import React from 'react';
import { useStateValue } from '../../store';
import Header from './Header.view';

const HeaderContainer = () => {
  const [{ name }] = useStateValue();
  return <Header name={name} />;
};

export default HeaderContainer;
