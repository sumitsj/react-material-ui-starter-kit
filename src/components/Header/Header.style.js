import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  logo: {
    textDecoration: 'underline',
  },
  logoLink: {
    color: '#fff',
  }
}));

export default useStyles;
