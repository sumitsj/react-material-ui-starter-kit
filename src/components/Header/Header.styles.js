import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  toolbar: {
    justifyContent: 'space-between',
  },
  logoLink: {
    color: '#fff',
  },
}));

export default useStyles;
