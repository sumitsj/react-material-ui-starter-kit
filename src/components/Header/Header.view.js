import React from "react";
import AppBar from "@material-ui/core/AppBar/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import useStyles from './Header.style';
import Link from "@material-ui/core/Link";

const Header = () => {
  const classes = useStyles();
  return <AppBar>
    <Toolbar>
      <Typography variant="h6" className={classes.logo}>
        <Link href="/" className={classes.logoLink}>
          Assisted HIP
        </Link>
      </Typography>
    </Toolbar>
  </AppBar>
};

export default Header;
