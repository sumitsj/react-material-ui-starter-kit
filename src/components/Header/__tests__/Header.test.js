import React from 'react';
import { render } from '@testing-library/react';
import Header from '..';

describe('Header', () => {
  describe('HTML Structure', () => {
    it('should render properly', () => {
      const { asFragment } = render(<Header />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
