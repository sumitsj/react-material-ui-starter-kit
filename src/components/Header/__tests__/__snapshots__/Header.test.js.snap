// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header HTML Structure should render properly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-logo-1 MuiTypography-h6"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-logoLink-2 MuiTypography-colorPrimary"
          href="/"
        >
          Assisted HIP
        </a>
      </h6>
    </div>
  </header>
</DocumentFragment>
`;
